pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/BytesTest.java

```diff
@@ -407,4 +407,48 @@ void matchesPrefixBytesTest() {
         RandomAccessData prefixBad3 = Bytes.wrap(new byte[]{0x01,0x02,0x03,0x04,0x05,0x06,0x07,0x08,0x09,0x00});
         assertFalse(primary.matchesPrefix(prefixBad3));
     }
+
+    @Test
+    void sortTests() {
+        var bytes1 = Bytes.wrap(new byte[0]);
+        var bytes2 = Bytes.wrap(new byte[0]);
+        assertEquals(0, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(0, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes1 = Bytes.wrap(new byte[]{7});
+        bytes2 = Bytes.wrap(new byte[]{7});
+        assertEquals(0, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(0, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes1 = Bytes.wrap(new byte[]{1});
+        bytes2 = Bytes.wrap(new byte[]{7});
+        assertEquals(0, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(-1, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes1 = Bytes.wrap(new byte[]{7});
+        bytes2 = Bytes.wrap(new byte[]{1});
+        assertEquals(0, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(1, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes1 = Bytes.wrap(new byte[]{7, 1});
+        bytes2 = Bytes.wrap(new byte[]{1});
+        assertEquals(-1, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(1, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes2 = Bytes.wrap(new byte[]{7, 2});
+        bytes1 = Bytes.wrap(new byte[]{1});
+        assertEquals(1, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(-1, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes1 = Bytes.wrap(new byte[]{7, 1});
+        bytes2 = Bytes.wrap(new byte[]{1, 7});
+        assertEquals(0, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(1, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes1 = Bytes.wrap(new byte[]{7, 2});
+        bytes2 = Bytes.wrap(new byte[]{7, 1});
+        assertEquals(0, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(1, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes1 = Bytes.wrap(new byte[]{1, 2});
+        bytes2 = Bytes.wrap(new byte[]{7, 1});
+        assertEquals(0, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(-1, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+        bytes1 = Bytes.wrap(new byte[]{7, 2});
+        bytes2 = Bytes.wrap(new byte[]{7, 2});
+        assertEquals(0, new Bytes.SORT_BY_LENGTH().compare(bytes1, bytes2));
+        assertEquals(0, new Bytes.SORT_BY_VALUE().compare(bytes1, bytes2));
+    }
```

Comment:
A great way to improve this code would be to break it into two parameterized tests (well, three tests if we have SORT_BY_LENGTH, SORT_BY_SIGNED_VALUE, SORT_BY_UNSIGNED_VALUE). JUnit5 has some great ways to do parameterized tests.

These below examples are just examples. They compile but not all the test pass (probably because the test is wrong).

This first example shows using the CSV Source parameterized test. It is easy to use when converting to certain primitive types:
```
    @ParameterizedTest
    @CsvSource(textBlock = """
            "", "", 0
            "a", "", 1
            "", "a", -1
            "a", "a", 0
            "aa", "a", 1
            "a", "aa", -1
            "ab", "ba", 0
            "abc", "cab", 0
            """)
    @DisplayName("Comparing Bytes by length")
    void compareByLength(String arr1, String arr2, int expected) {
        var bytes1 = Bytes.wrap(arr1);
        var bytes2 = Bytes.wrap(arr2);
        assertEquals(expected, Bytes.SORT_BY_LENGTH.compare(bytes1, bytes2));
    }
```

These next examples show using `MethodSource` style. Both CSVSource and MethodSource can be found online with lots of docs on how to use them:

```
    @ParameterizedTest
    @MethodSource
    @DisplayName("Comparing Bytes by signed value")
    void compareBySignedBytes(byte[] arr1, byte[] arr2, int expected) {
        var bytes1 = Bytes.wrap(arr1);
        var bytes2 = Bytes.wrap(arr2);
        assertEquals(expected, Bytes.SORT_BY_SIGNED_VALUE.compare(bytes1, bytes2));
    }

    static Stream<Arguments> compareBySignedBytes() {
        return Stream.of(
                Arguments.of(new byte[0], new byte[0], 0),
                Arguments.of(new byte[0], new byte[]{1}, -1),
                Arguments.of(new byte[]{1}, new byte[0], 1),
                Arguments.of(new byte[]{1}, new byte[]{2}, 1),
                Arguments.of(new byte[]{2}, new byte[]{1}, -1),
                Arguments.of(new byte[]{-1}, new byte[]{2}, 1),
                Arguments.of(new byte[]{2}, new byte[]{-1}, -1),
                Arguments.of(new byte[]{-1}, new byte[]{-2}, -1),
                Arguments.of(new byte[]{-2}, new byte[]{-1}, 1)
        );
    }

    @ParameterizedTest
    @MethodSource
    @DisplayName("Comparing Bytes by unsignedValue")
    void compareByUnsignedBytes(byte[] arr1, byte[] arr2, int expected) {
        var bytes1 = Bytes.wrap(arr1);
        var bytes2 = Bytes.wrap(arr2);
        assertEquals(expected, Bytes.SORT_BY_UNSIGNED_VALUE.compare(bytes1, bytes2));
    }

    static Stream<Arguments> compareByUnsignedBytes() {
        return Stream.of(
                Arguments.of(new byte[0], new byte[0], 0),
                Arguments.of(new byte[0], new byte[]{1}, -1),
                Arguments.of(new byte[]{1}, new byte[0], 1),
                Arguments.of(new byte[]{1}, new byte[]{2}, 1),
                Arguments.of(new byte[]{2}, new byte[]{1}, -1),
                Arguments.of(new byte[]{-1}, new byte[]{2}, 1),
                Arguments.of(new byte[]{2}, new byte[]{-1}, -1),
                Arguments.of(new byte[]{-1}, new byte[]{-2}, -1),
                Arguments.of(new byte[]{-2}, new byte[]{-1}, 1)
        );
    }
```

What is nice about using parameterized tests is that they all run individually and those that pass and fail are listed separately.